src/net_processing.cpp

```diff
@@ -111,6 +111,8 @@ static constexpr unsigned int INVENTORY_BROADCAST_MAX = 7 * INVENTORY_BROADCAST_
 static constexpr unsigned int AVG_FEEFILTER_BROADCAST_INTERVAL = 10 * 60;
 /** Maximum feefilter broadcast delay after significant change. */
 static constexpr unsigned int MAX_FEEFILTER_CHANGE_DELAY = 5 * 60;
+/** Average delay between rebroadcasts */
+static const std::chrono::seconds TX_REBROADCAST_INTERVAL = std::chrono::seconds{60 * 60};
```

Comment:
are you proposing the minimum to support a user configurable frequency, or for the current case (since its possible bc of poisson distribution)?

Reply:
would be happy to support customizable rebroadcast interval in a future PR if that's of interest. request for anyone coming across this who'd like that feature to leave a 👍response

Reply:
I think the min can be hard-coded, and that it shouldn't happen more frequently even when drawing poisson time (e.g.,  something like: min_time + poisson(avg_time - min_time)).

So as to impose a strict cap on how frequently this is happening, because if you draw out of a poisson a large number of times eventually you'll see a run of small interval windows.

You also want something to ensure that there is always some entropy, even if min_time == avg_time (maybe enforce that avg_time >= min_time*2).

Then, I think that the *actual* rebroadcast interval should be configurable, based on a user parameter. E.g., default is 30 minutes, but can be changes down to the min_time. This can happen in a future PR.

Reply:
I don't think a customizable rebroadcast interval is a good idea:

- we should avoid adding command line options that have minimal impact on user experience. Take a look at the 10s of options in `init.cpp`. I'm sure there are many of them that hardly anyone uses, and they just add unnecessary complexity/interactions to the code.
- we should try to avoid adding unnecessary fingerprints to nodes by keeping their p2p behaviour as uniform as possible.

Reply:
Then that's an argument against doing this *at all*, as it increases bandwidth requirements for low-resourced nodes who need to rebroadcast transactions.

Reply:
To be clear; I'm in favor of this change :)

I just think if you're a low resourced node you'd want to be able to decrease the frequency at which your rebroadcasts are occurring if you have to send out an entire block worth of data.

Reply:
I think there's a case for reviewing the worst case resource usage, but I think in the normal case the bandwidth for rebroadcast will be very low and not really something that needs to be throttled even for low powered nodes. (Reducing your mempool size would be one way of reducing your outgoing rebroadcast traffic, but I think that's mostly just a tradeoff against incoming rebroadcast traffic)

Reply:
Reducing mempool does not obviously help with resource reduction for a few reasons:

1) because of compact blocks (slower to get new blocks, so if you're peak-bandwidth capped it's bad).
2) If you're bandwidth constrained, you aren't necessarily memory constrained either (why throw away a transaction if it seems good?)
3) It stands to reason that when fees are going up, you'll evict a lot of stuff from your mempool as newer txns go in, and when the fees go back down you'll need stuff re-relayed.


I think I'd just like to understand how this does impact bandwidth usage, it seemed to not be a non trivial amount more as it's 1/f * block size * period * #peers more data volume? If it's actually small then so be it. But it does seem, to me, to be non-negligible

Reply:
Seems like the customizable rebroadcast interval is a proposed solution for concerns around bandwidth usage for low resourced nodes. 

I agree that we should ensure these rebroadcast changes don’t impose a significant bandwidth requirement.

I believe the current changes require little additional bandwidth, which my initial monitoring from running this patch supports. More info posted in the main comment [thread](https://github.com/bitcoin/bitcoin/pull/16698#issuecomment-570838774). 

But this is not conclusive for determining that bandwidth impact is low in all conditions. If further monitoring leads to deciding we need to restrict resource usage further, we have options. For the fingerprint reasons mentioned above I don't think a customizable rebroadcast interval makes the most sense. However, we could consider a kill switch to disable rebroadcast all together (along the lines of current implementation walletbroadcast=0). Also available (and probably the first line of experimentation) would be tinkering with the params to tighten the txns selected for rebroadcast.

I’ve updated my [gist](https://gist.github.com/amitiuttarwar/b592ee410e1f02ac0d44fcbed4621dba#open-questions--solutions) with this open questions & possible solutions and moving forward will maintain it with the options & latest thinking. 

Since this comment thread is about customizable rebroadcast intervals, I'm going to resolve the conversation. But let's continue to discuss bandwidth impact in the main PR thread as I continue to monitor & share data. 

@JeremyRubin let me know what you think. esp if you disagree or have any questions.